src/gpl/src/nesterovBase.cpp

```diff
@@ -602,12 +602,21 @@ void BinGrid::initBins()
   int64_t totalBinArea
       = static_cast<int64_t>(ux_ - lx_) * static_cast<int64_t>(uy_ - ly_);
 
-  int64_t averagePlaceInstArea
-      = pb_->placeInstsArea() / pb_->placeInsts().size();
+  int64_t averagePlaceInstArea = 0;
+  if (!pb_->placeInsts().empty()) {
+    averagePlaceInstArea = pb_->placeInstsArea() / pb_->placeInsts().size();
+  }
+
+  int64_t idealBinArea = 0;
```

Comment:
Is this a situation we want to recover from or send an error to the user? Separately, should placeInsts ever be empty at this point?

Reply:
If there are no instances to place then we should skip all of gpl. Does this design really have no instances to place?

Reply:
I do see instances placed at the beginning of GPL with debug mode. I will investigate further. Either way, I think the checking I have added so far does not hurt.